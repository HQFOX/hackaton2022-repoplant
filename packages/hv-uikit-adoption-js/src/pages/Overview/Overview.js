import React from "react";
import PropTypes from "prop-types";
import HvTypography from "@hv/uikit-react-core/dist/Typography";
import HvGrid from "@hv/uikit-react-core/dist/Grid";
import { useTranslation } from "react-i18next";
import withLayout from "lib/hocs/withLayout";
import AssetInventory from "components/overview/AssetInventory";

const Overview = ({ classes }) => {
  const { t } = useTranslation();

  return (
    <HvGrid container>
      <HvGrid item xl={12}>
        <HvTypography variant="3xlTitle" className={classes.title}>
          {t("pages.overview.title")}
        </HvTypography>
      </HvGrid>
      <HvGrid item xl={12}>
        <AssetInventory />
      </HvGrid>
    </HvGrid>
  );
};

Overview.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired
};

export default withLayout(Overview);
