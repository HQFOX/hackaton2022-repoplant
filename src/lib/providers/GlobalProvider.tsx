import React from "react";
import { HashRouter as Router } from "react-router-dom";
import { ApolloClient, InMemoryCache, ApolloProvider } from "@apollo/client";

import { ThemeProvider } from "lib/context/ThemeContext";
import { NavigationProvider } from "lib/context/NavigationContext";
import { AuthProvider } from "lib/context/AuthContext";
import navigation from "lib/navigation";

const client = new ApolloClient({
  uri: "https://swapi-graphql.netlify.app/.netlify/functions/index",
  cache: new InMemoryCache(),
  credentials: "omit",
});

const GlobalProvider: React.FC = ({ children }) => (
  <Router basename={process.env.PUBLIC_URL}>
    <ApolloProvider client={client}>
      <NavigationProvider navigation={navigation}>
        <AuthProvider>
          <ThemeProvider>{children}</ThemeProvider>
        </AuthProvider>
      </NavigationProvider>
    </ApolloProvider>
  </Router>
);

export default GlobalProvider;
