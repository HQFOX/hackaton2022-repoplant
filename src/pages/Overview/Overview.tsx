import React from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";
import {
  HvGrid,
  HvCard,
  HvCardHeader,
  HvCardMedia,
  HvActionBar,
  HvActionsGeneric,
} from "@hitachivantara/uikit-react-core";

import { Container } from "components/layout";
import { ReactComponent as SWLogo } from "assets/star-wars-logo.svg";
import characters from "assets/characters.jpg";
import starships from "assets/starships.jpg";

import useStyles from "./styles";

const Overview: React.FC = () => {
  const { t } = useTranslation("overview");
  const classes = useStyles();
  const history = useHistory();

  return (
    <Container>
      <HvGrid container alignItems="center" justifyContent="center">
        <HvGrid container item xs={12} justifyContent="center">
          <SWLogo className={classes.logo} />
        </HvGrid>
        <HvGrid item xs={12} sm={5} md={5}>
          <HvCard bgcolor="atmo1" className={classes.card}>
            <HvCardHeader title={t("charactersCard.title")} />
            <HvCardMedia image={characters} className={classes.img} />
            <HvActionBar>
              <HvActionsGeneric
                actions={[
                  {
                    id: "exploreCharacters",
                    label: t("charactersCard.button"),
                  },
                ]}
                actionsCallback={() => history.push(`/star-wars/characters`)}
              />
            </HvActionBar>
          </HvCard>
        </HvGrid>
        <HvGrid item xs={12} sm={5} md={5}>
          <HvCard
            bgcolor="atmo1"
            className={classes.card}
            onClick={() => history.push(`/star-wars/starships`)}
          >
            <HvCardHeader title={t("starshipsCard.title")} />
            <HvCardMedia image={starships} className={classes.img} />
            <HvActionBar>
              <HvActionsGeneric
                actions={[
                  {
                    id: "exploreStarships",
                    label: t("starshipsCard.button"),
                  },
                ]}
                actionsCallback={() => history.push(`/star-wars/starships`)}
              />
            </HvActionBar>
          </HvCard>
        </HvGrid>
      </HvGrid>
    </Container>
  );
};

export default Overview;
