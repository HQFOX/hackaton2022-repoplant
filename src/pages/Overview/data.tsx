import { gql } from "@apollo/client";

export const QUERY_GET_SPACESHIPS = gql`
  query GetSpaceships {
    allStarships {
      edges {
        node {
          id
          name
          model
          passengers
          crew
          length
          cargoCapacity
        }
      }
    }
  }
`;
