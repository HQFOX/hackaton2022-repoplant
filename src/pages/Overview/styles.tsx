import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  card: {
    cursor: "pointer",
    height: 150,
  },
}));

export default useStyles;
