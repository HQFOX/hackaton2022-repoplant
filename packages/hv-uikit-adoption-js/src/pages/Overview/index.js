import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import Overview from "./Overview";

export default withStyles(styles, { name: "Overview", withTheme: true })(
  Overview
);
