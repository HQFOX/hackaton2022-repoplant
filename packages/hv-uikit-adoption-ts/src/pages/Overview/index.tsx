import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import Overview from "./Overview";

export type OverviewProps = WithStyles<typeof styles>;

export default withStyles(styles, { name: "Overview", withTheme: true })(
  Overview
);
