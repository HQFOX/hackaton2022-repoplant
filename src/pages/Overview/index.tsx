import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
import { gql, useQuery, QueryResult } from "@apollo/client";

import styles from "./styles";
import Overview from "./Overview";

export function useSpaceships(): QueryResult {
  return useQuery(gql`
    query GetSpaceships {
      allStarships {
        edges {
          node {
            id
            name
            model
            passengers
            crew
            length
            cargoCapacity
          }
        }
      }
    }
  `);
}

export type OverviewProps = WithStyles<typeof styles>;

export default withStyles(styles, { name: "Overview" })(Overview);
