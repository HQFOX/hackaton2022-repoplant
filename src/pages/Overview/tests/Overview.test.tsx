import React from "react";
import { render, screen } from "@testing-library/react";

import GlobalProvider from "lib/providers/GlobalProvider";
import Overview from "../Overview";

const renderWithProvider = () =>
  render(
    <GlobalProvider>
      <Overview />
    </GlobalProvider>
  );

describe("Overview", () => {
  test("should match the snapshot", () => {
    const { container } = renderWithProvider();
    expect(container).toMatchSnapshot();
  });

  test("should render cards title", () => {
    renderWithProvider();
    const charactersCardTitle = screen.getByText(/charactersCard.title/i);
    const starshipsCardTitle = screen.getByText(/starshipsCard.title/i);

    expect(charactersCardTitle).toBeInTheDocument();
    expect(starshipsCardTitle).toBeInTheDocument();
  });
});
