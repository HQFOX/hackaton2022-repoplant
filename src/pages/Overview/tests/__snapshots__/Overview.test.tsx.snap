// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should match the snapshot 1`] = `
<div>
  <div
    class="makeStyles-content-4"
  >
    <main
      class="MuiContainer-root HvContainer-root makeStyles-container-5 makeStyles-hasSecondLevel-6"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
        >
          <svg
            class="makeStyles-logo-1"
          >
            star-wars-logo.svg
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
        >
          <div
            class="MuiBox-root MuiBox-root-8 HvIsCardGridElement HvCard-root makeStyles-card-2"
          >
            <div
              class="HvCard-semanticContainer"
            >
              <div
                class="HvCard-semanticBar HvCard-sema0"
              />
              <div
                class="HvCard-icon"
              />
            </div>
            <div
              class="MuiCardHeader-root HvCardHeader-root"
            >
              <div
                class="MuiCardHeader-content HvCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title HvCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  charactersCard.title
                </span>
              </div>
            </div>
            <div
              class="MuiCardMedia-root HvCardMedia-root makeStyles-img-3"
              role="img"
              style="background-image: url(characters.jpg);"
            />
            <div
              class="HvActionBar-root"
            >
              <div
                class="HvActionsGeneric-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvActionsGeneric-button HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                  id="exploreCharacters"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    charactersCard.button
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
        >
          <div
            class="MuiBox-root MuiBox-root-9 HvIsCardGridElement HvCard-root makeStyles-card-2"
          >
            <div
              class="HvCard-semanticContainer"
            >
              <div
                class="HvCard-semanticBar HvCard-sema0"
              />
              <div
                class="HvCard-icon"
              />
            </div>
            <div
              class="MuiCardHeader-root HvCardHeader-root"
            >
              <div
                class="MuiCardHeader-content HvCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title HvCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  starshipsCard.title
                </span>
              </div>
            </div>
            <div
              class="MuiCardMedia-root HvCardMedia-root makeStyles-img-3"
              role="img"
              style="background-image: url(starships.jpg);"
            />
            <div
              class="HvActionBar-root"
            >
              <div
                class="HvActionsGeneric-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text HvActionsGeneric-button HvButton-ghostSVG MuiButton-textPrimary HvButton-ghost"
                  id="exploreStarships"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    starshipsCard.button
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
